code/datums/riding.dm

```diff
@@ -264,19 +269,75 @@
 			buckled_mob.setDir(ridden.dir)
 			ridden.pixel_x = -48
 			ridden.pixel_y = -48
-			switch(ridden.dir)
-				if(NORTH)
-					buckled_mob.pixel_x = -10
-					buckled_mob.pixel_y = -3
-				if(SOUTH)
-					buckled_mob.pixel_x = 16
-					buckled_mob.pixel_y = 3
-				if(EAST)
-					buckled_mob.pixel_x = -4
-					buckled_mob.pixel_y = 30
-				if(WEST)
-					buckled_mob.pixel_x = 4
-					buckled_mob.pixel_y = -1
+			if(buckled_mob = ridden.buckled_mobs[1])
```

Comment:
no thanks.
give ```/datum/riding``` a proc, ```/datum/riding/proc/passanger_index_to_pixel_offsets()``` and then we can generalize this to all vehicles.

```passanger_index_to_pixel_offsets()``` would return ```list("[NORTH]" = list("x" = 0, "y" = 0), "[SOUTH]" = list("x" = 0, "y" = 0), ...)```